tests/ui/perfherder/graphs-view/test_data_modal_test.jsx

```diff
@@ -0,0 +1,193 @@
+  const tagOccurency = tags.filter((tag) => tag === 'repeated-tag').length;
+
+  expect(tagOccurency).toBeGreaterThan(1);
+
+  const tag = await waitFor(() => getAllByTestId('available-tag repeated-tag'));
+
+  expect(tag).toHaveLength(1);
+});
+
+test('Selecting two tags from tags multi select shows the tests that have both tags', async () => {
+  const { queryByTestId, getByText, queryAllByTestId } = testDataModel();
+  const activeTag1 = seriesDataWithRepeatedTag[0].tags[0];
+  const activeTag2 = seriesDataWithRepeatedTag[0].tags[1];
+  const activeTags = [activeTag1, activeTag2];
+  const taggedTests = seriesDataWithRepeatedTag.filter((test) =>
+    activeTags.every((activeTag) => test.tags.includes(activeTag)),
+  );
+  let tests = await waitFor(() => queryByTestId('tests'));
+
+  expect(tests).toHaveLength(seriesDataWithRepeatedTag.length);
+
+  const tag1 = await waitFor(() => getByText(activeTag1));
+  const tag2 = await waitFor(() => getByText(activeTag2));
+
+  fireEvent.click(tag1);
+  fireEvent.click(tag2);
+
+  expect(queryAllByTestId(/active-tag/)).toHaveLength(activeTags.length);
+
+  tests = await waitFor(() => queryByTestId('tests'));
+
+  expect(tests).toHaveLength(taggedTests.length);
+});
+
+test('Selecting a tag from tags multi select shows the tests that have the specific tag', async () => {
+  const { queryByTestId, getByText, getByTestId } = testDataModel();
+  const tag = await waitFor(() => getByText(seriesData[0].tags[0]));
+
+  fireEvent.click(tag);
+
+  const activeTag = await waitFor(() =>
+    getByTestId(`active-tag ${seriesData[0].tags[0]}`),
+  );
+
+  expect(activeTag).toBeInTheDocument();
+
+  const tests = await waitFor(() => queryByTestId('tests'));
+  const test = await waitFor(() => queryByTestId(seriesData[0].id.toString()));
+
+  expect(tests).toContain(test);
+});
+
+test("Selecting 'all tags' from tags multi select makes all the available tags active", async () => {
+  const { getAllByTestId, getByText } = testDataModel();
+  const availableTags = await waitFor(() => getAllByTestId(/available-tag/));
+  const allTags = await waitFor(() => getByText('all tags'));
+
+  fireEvent.click(allTags);
+
+  const activeTags = await waitFor(() => getAllByTestId(/active-tag/));
+
+  expect(activeTags).toHaveLength(availableTags.length - 1);
+});
+
+test('Active tag can be deactivated by clicking on it from multi select', async () => {
+  const { queryByTestId, getByText, getByTestId } = testDataModel();
+  const tag = await waitFor(() => getByText(seriesData[0].tags[0]));
+
+  fireEvent.click(tag);
+
+  let activeTag = await waitFor(() =>
+    getByTestId(`active-tag ${seriesData[0].tags[0]}`),
+  );
+
+  expect(activeTag).toBeInTheDocument();
+
+  fireEvent.click(tag);
+
+  activeTag = await waitFor(() =>
+    queryByTestId(`active-tag ${seriesData[0].tags[0]}`),
+  );
+
+  expect(activeTag).toBeNull();
+});
+
+test('Active tag can be deactivated by clicking on it from active tags list', async () => {
```

Comment:
I don't see any difference between this test & the previous one.

Reply:
The difference is that an `active-tag` can be deactivated from the available tags list and from the active tags list. The tests look similar at first and after `expect(activeTag).toBeInTheDocument();` there is a difference.
In this test we have `fireEvent.click(activeTag)` - `activeTag `is from the active tags list.
In the previous test we have `fireEvent.click(tag)` - `tag `is from the available tags list.

I made some changes for the test name in the previous test. What do you think? How could we improve this?

Reply:
Ok, got it. Instead of simply `tag`, maybe use `availableTag` instead?